contents/tutorials/flutter-ab-tests.md

```diff
@@ -0,0 +1,358 @@
+  alt="Events captured in PostHog" 
+  classes="rounded"
+/>
+
+## 4. Create an A/B test in PostHog
+
+If you haven't done so already, you'll need to [upgrade](https://us.posthog.com/organization/billing) your PostHog account to include A/B testing. This requires entering your credit card, but don't worry, we have a [generous free tier](/pricing) of 1 million requests per month – so you won't be charged anything yet.
+
+Next, go to the [A/B testing tab](https://us.posthog.com/experiments) and create an A/B test by clicking the **New experiment** button. Add the following details to your experiment:
+
+1. Name it "My cool experiment".
+2. Set "Feature flag key" to `my-cool-experiment`.
+3. Under the experiment goal, select the `feature_button_clicked` event we created in the previous step.
+4. Use the default values for all other fields.
+
+Click "Save as draft" and then click "Launch".
+
+<ProductScreenshot
+  imageLight={TestSetupLight} 
+  imageDark={TestSetupDark} 
+  alt="Experiment setup in PostHog" 
+  classes="rounded"
+/>
+
+## 5. Implement the A/B test code
+
+The final step is to add the experiment code. We'll add code that does the following:
+
+1. Fetch the `my-cool-experiment` flag using [`await Posthog().getFeatureFlag('my-cool-experiment')`](/docs/libraries/flutter#multivariate-feature-flags).
+2. Change the background color of `FeatureScreenView` based on the value of the flag (`control` or `test`).
+
+To do this, update the code in `main.dart` to the following:
+
+```dart
+import 'package:flutter/material.dart';
+import 'feature_screen_view.dart';
+import 'package:posthog_flutter/posthog_flutter.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: 'Flutter A/B Test App',
+      home: MainScreen(),
+    );
+  }
+}
+
+class MainScreen extends StatelessWidget {
+  const MainScreen({super.key});
+
+  void _getFeatureFlagAndNavigate(BuildContext context) {
+    _fetchFeatureFlagValue((isTestVariant) {
+      Navigator.push(
+        context,
+        MaterialPageRoute(builder: (context) => FeatureScreenView(isTestVariant: isTestVariant)),
+      );
+    });
+  }
+
+ void _fetchFeatureFlagValue(Function(bool) onFlagFetched) async {
+    String featureFlagValue = await Posthog().getFeatureFlag('my-cool-experiment') as String;
+    bool isTestVariant = featureFlagValue == 'test';
+    onFlagFetched(isTestVariant);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Main Screen')),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            const Text('Hello, world!'),
+            ElevatedButton(
+              child: const Text('Go to Next Screen'),
+              onPressed: () => _getFeatureFlagAndNavigate(context),
```

Comment:
Its better to make `onPressed` async and `await` for the method since its also async

Reply:
I tried this but got a lint warning/error, hence moved to separated function

Reply:
Not sure why it'd be a warning error?
did you do this?

```
onPressed: () async {
                      await _getFeatureFlagAndNavigate(context);
},
```

Reply:
Here's the warning I kept getting when trying to put everything in one function:

<img width="834" alt="Screenshot 2024-03-01 at 2 26 51 PM" src="https://github.com/PostHog/posthog.com/assets/7090054/2c0a6121-2062-4afa-9884-e5e04e55ef37">

Reply:
That worked! Managed to clean it up now. Thanks!